libraries/botbuilder/src/channelServiceHandler.ts

```diff
@@ -597,27 +598,26 @@ export class ChannelServiceHandler {
      * @private
      */
     private async authenticate(authHeader: string): Promise<ClaimsIdentity> {
-        try {
-            if (!authHeader) {
-                const isAuthDisable = this.credentialProvider.isAuthenticationDisabled();
-                if (isAuthDisable) {
-                    // In the scenario where Auth is disabled, we still want to have the
-                    // IsAuthenticated flag set in the ClaimsIdentity. To do this requires
-                    // adding in an empty claim.
-                    return new ClaimsIdentity([], false);
-                }
+        if (!authHeader) {
+            const isAuthDisabled = await this.credentialProvider.isAuthenticationDisabled();
+            if (!isAuthDisabled) {
+                throw new StatusCodeError(StatusCodes.UNAUTHORIZED);
             }
 
-            return await JwtTokenValidation.validateAuthHeader(
-                authHeader,
-                this.credentialProvider,
-                this.channelService,
-                'unknown',
-                undefined,
-                this.authConfig
-            );
-        } catch (err) {
-            throw new StatusCodeError(StatusCodes.UNAUTHORIZED);
+            // In the scenario where Auth is disabled, we still want to have the
+            // IsAuthenticated flag set in the ClaimsIdentity. To do this requires
+            // adding in an empty claim.
+            // Since ChannelServiceHandler calls are always a skill callback call, we set the skill claim too.
+            return SkillValidation.createAnonymousSkillClaim();
         }
+
+        return JwtTokenValidation.validateAuthHeader(
```

Comment:
good cleanup, thanks! We should probably evaluate adding a linting rule for "no await on return"